File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter02/CPP/totalSalary.cpp
/*Libro:Fundamentos de Programación, Algoritmos, estructura de datos y objetos 4ta edición. 
    Capitulo 02.-Metodología de la programación y desarrollo de software.
    Problema 14.Escribir un algoritmo que lea el nombre de un empleado, las horas trabajadas, 
    el precio por hora y calcule los impuestos a pagar (tasa = 25%) y el salario neto.
*/
#include<iostream>
using namespace std;
int main(){
    char nameWorker[120]; //arreglo que almacenara el nombre del trabajador
    int hoursWorked;    //Horas trabajadas
    int pricePerHour;   //Pago por Hora
    int tax=25;         //impuesto a decucir 25%
    int grossSalary;    //Salario bruto
    int netSalary;      //salrio neto (salario despues de descuento de impuestos)
    int salaryDiscount; //Impuesto a descontar

    cout<<"Ingrese el nombre del trabajador: ";
    cin>>nameWorker;

    cout<<"Ingresa las horas trabajadas: ";
    cin>>hoursWorked;

    cout<<"Ingresa el precio de la hora trabajada $: ";
    cin>>pricePerHour;

    grossSalary=hoursWorked*pricePerHour;
    
    salaryDiscount=(grossSalary*tax)/100;

    netSalary=grossSalary-salaryDiscount;

    cout<<"\nEmpleado: "<<nameWorker<<"\nSueldo neto $: "<<netSalary<<"\n";

    return 0;
}  

File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter02/JAVASCRIPT/js/totalSalary.js
function calculate(){
    let nameWorker;
    let hoursWorked; 
    let pricePerHour;

    let tax=25;         //impuesto a decucir 25%
    let grossSalary;    //Salario bruto
    let netSalary;      //salrio neto (salario despues de descuento de impuestos)
    let salaryDiscount; //Impuesto a descontar


    let paragrafg=document.createElement('p');
    paragrafg.setAttribute("id", "idParagrafg");
        
    if(document.getElementById("idBtnCalculate").value=="Calcular"){
            
            nameWorker=document.getElementById("idName").value;
            hoursWorked=parseInt(document.getElementById("idHoursWorked").value);
            pricePerHour=parseInt(document.getElementById("idPriceHour").value);
            
            grossSalary=hoursWorked*pricePerHour;
            salaryDiscount=(grossSalary*tax)/100;
            netSalary=grossSalary-salaryDiscount;

            paragrafg.innerHTML+="Empleado: "+nameWorker+"<p></p>Salario Bruto: "+grossSalary+"<p></p>Impuesto a descontar: "+tax+"%<p></p>Salario Neto: "+netSalary;

            document.getElementById("idSection").appendChild(paragrafg);
            document.getElementById("idBtnCalculate").value="Limpiar";
        }else{
            document.getElementById("idName").value="";
            document.getElementById("idHoursWorked").value="";
            document.getElementById("idPriceHour").value="";
            document.getElementById("idParagrafg").remove();
        
            document.getElementById("idBtnCalculate").value="Calcular";
        }
    

}